src/components/ec-navigation-link/types.ts

```diff
@@ -0,0 +1,15 @@
+import type { Ref } from 'vue';
+
+import type { IconName } from '../ec-icon/icon-names';
+
+export interface NavigationLinkProps {
+  text: string,
+  iconName: IconName,
+  iconSize?: number,
+  url: string,
+  isRouterLink?: boolean,
+  isActive?: boolean | Ref<boolean>,
```

Comment:
there is a type `MaybeRef` in vue for this case. It's also worth considering `MaybeRefOrGetter`